epochStart/shardchain/trigger.go

```diff
@@ -283,42 +292,49 @@ func (t *trigger) ReceivedHeader(header data.HeaderHandler) {
 		return
 	}
 
-	hdrHash, err := core.CalculateHash(t.marshalizer, t.hasher, metaHdr)
-	if err != nil {
+	if _, ok = t.mapHashHdr[string(metaBlockHash)]; ok {
 		return
 	}
-
-	if _, ok = t.mapHashHdr[string(hdrHash)]; ok {
-		return
-	}
-	if _, ok = t.mapEpochStartHdrs[string(hdrHash)]; ok {
+	if _, ok = t.mapEpochStartHdrs[string(metaBlockHash)]; ok {
 		return
 	}
 
-	t.updateTriggerFromMeta(metaHdr, hdrHash)
+	t.updateTriggerFromMeta(metaHdr, metaBlockHash)
 }
 
 // call only if mutex is locked before
 func (t *trigger) updateTriggerFromMeta(metaHdr *block.MetaBlock, hdrHash []byte) {
 	if metaHdr.IsStartOfEpochBlock() {
 		t.newEpochHdrReceived = true
 		t.mapEpochStartHdrs[string(hdrHash)] = metaHdr
+	}
+
+	t.mapHashHdr[string(hdrHash)] = metaHdr
+	t.mapNonceHashes[metaHdr.Nonce] = append(t.mapNonceHashes[metaHdr.Nonce], string(hdrHash))
 
-	} else {
-		t.mapHashHdr[string(hdrHash)] = metaHdr
-		t.mapNonceHashes[metaHdr.Nonce] = append(t.mapNonceHashes[metaHdr.Nonce], string(hdrHash))
+	sortedMetaInfo := make([]*metaInfo, 0, len(t.mapEpochStartHdrs))
+	for hash, hdr := range t.mapEpochStartHdrs {
+		metaInfo := &metaInfo{
+			hdr:  hdr,
+			hash: hash,
+		}
+		sortedMetaInfo = append(sortedMetaInfo, metaInfo)
 	}
 
-	for hash, meta := range t.mapEpochStartHdrs {
-		canActivateEpochStart, finalityAttestingRound := t.checkIfTriggerCanBeActivated(hash, meta)
-		if canActivateEpochStart && t.epoch < meta.Epoch {
-			t.epoch = meta.Epoch
+	sort.Slice(sortedMetaInfo, func(i, j int) bool {
+		return sortedMetaInfo[i].hdr.Epoch < sortedMetaInfo[j].hdr.Epoch
+	})
+
+	for _, metaInfo := range sortedMetaInfo {
```

Comment:
can you rename one of the `metaInfo` so the warning with shadow declaration is removed?